src/test/java/uk/gov/pay/ledger/pact/RefundSucceededEventQueueContractTest.java

```diff
@@ -0,0 +1,97 @@
+    @Rule
+    public MessagePactProviderRule mockProvider = new MessagePactProviderRule(this);
+
+    @Rule
+    public AppWithPostgresAndSqsRule appRule = new AppWithPostgresAndSqsRule(
+            config("queueMessageReceiverConfig.backgroundProcessingEnabled", "true")
+    );
+
+    private byte[] currentMessage;
+    private String resourceExternalId = RandomStringUtils.randomAlphanumeric(14);
+    private String parentResourceExternalId = RandomStringUtils.randomAlphanumeric(14);
+    private String reference = RandomStringUtils.randomAlphanumeric(14);
+    private ZonedDateTime eventDate = ZonedDateTime.parse("2019-07-25T10:45:46.391654Z");
+
+    @Pact(provider = "connector", consumer = "ledger")
+    public MessagePact createRefundSucceededEventPact(MessagePactBuilder builder) {
+        QueueRefundEventFixture refundSucceededEventFixture = aQueueRefundEventFixture()
+                .withResourceExternalId(resourceExternalId)
+                .withParentResourceExternalId(parentResourceExternalId)
+                .withEventDate(eventDate)
+                .withEventType("REFUND_SUCCEEDED")
+                .withRefundedBy(null)
+                .withReference(reference)
+                .withDefaultEventDataForEventType("REFUND_SUCCEEDED");
+
+        Map<String, String> metadata = new HashMap();
+        metadata.put("contentType", "application/json");
+
+        PactDslJsonBody pactBody = refundSucceededEventFixture.getAsPact();
+
+        return builder
+                .expectsToReceive("a refund succeeded message")
+                .withMetadata(metadata)
+                .withContent(pactBody)
+                .toPact();
+    }
+
+    @Test
+    @PactVerification({"connector"})
+    public void test() throws Exception {
+        aTransactionFixture()
+                .withExternalId(parentResourceExternalId)
+                .insert(appRule.getJdbi());
+
+        appRule.getSqsClient().sendMessage(SqsTestDocker.getQueueUrl("event-queue"), new String(currentMessage));
+
+        DatabaseTestHelper dbHelper = DatabaseTestHelper.aDatabaseTestHelper(appRule.getJdbi());
+        await().atMost(1, TimeUnit.SECONDS).until(
+                () -> {
+                    Map<String, Object> event;
+                    try {
+                        event = dbHelper.getEventByExternalId(resourceExternalId);
+                        String eventReference = new JsonParser().parse(event.get("event_data").toString())
+                                .getAsJsonObject().get("reference").getAsString();
+                        return (resourceExternalId.equals(event.get("resource_external_id")) &&
+                                parentResourceExternalId.equals(event.get("parent_resource_external_id")) &&
+                                reference.equals(eventReference));
+                    } catch (NoSuchElementException e) {
+                        return false;
+                    }
+                }
+        );
+    }
```

Comment:
This https://github.com/alphagov/pay-ledger/blob/master/src/test/java/uk/gov/pay/ledger/pact/PaymentCreatedEventQueueContractTest.java#L62-L80 looks like slightly cleaner way of writing the test (to me - I've written it). Would it be possible for you to update it (and other pact tests) in this way?

Reply:
Updated to use `TransactionEntity`